### ISSUE TITLE
when i  am adding new product or modifying the record it is not showing the result set in search 

### ISSUE THREAD

**Issue by pawanvirsingh** [django-oscar/django-oscar#2599]
As I used Solr for search backend, and  once my product get index and now the second time when I add new  products and change price after index created that  product not being search in  and sorting of price also not worked for new change ( i need to rebuild index then start the java server then will work properly  ), is there any other solution  for the same . I am using Solr 4.7.2.

**Comment by sasha0** [comment]
Currently Oscar does not have automatic search index update on related model change. You can integrate Haystack's `RealtimeSignalProcessor` or implement search index record on your own.

**Comment by pawanvirsingh** [comment]
yes I tried but some time it worked some time not .

i tried this in signals.py 

from oscar.apps.partner.models import StockRecord
from .models import Product

from django.db import models
from haystack import signals


class UserOnlySignalProcessor(signals.BaseSignalProcessor):
    def setup(self):
    	print "managal"
        # Listen only to the ``User`` model.
        models.signals.post_save.connect(self.handle_save, sender=Product)
        models.signals.post_delete.connect(self.handle_delete, sender=Product)
        models.signals.post_save.connect(self.handle_save, sender=StockRecord)
        models.signals.post_delete.connect(self.handle_delete, sender=StockRecord)

    def teardown(self):
    	print "dangaldsfsdf"
        # Disconnect only for the ``User`` model.
        models.signals.post_save.disconnect(self.handle_save, sender=Product)
        models.signals.post_delete.disconnect(self.handle_delete, sender=Product)
        models.signals.post_save.disconnect(self.handle_save, sender=StockRecord)
        models.signals.post_delete.disconnect(self.handle_delete, sender=StockRecord)

it works some time sometime not

**Comment by tejbad** [comment]
I am also facing same issue. Please tell me if you have found any solution for this.

**Comment by Bastilla123** [comment]
Yeah this is a mega bug. Haystack without sync between normal postgres datbase and solr is a holy shit. Sorry unusable...
